splice_machine/src/main/java/com/splicemachine/derby/stream/utils/ExternalTableUtils.java

```diff
@@ -261,16 +262,25 @@ public static void preSortColumns(StructField[] schema, int[] partitionColumnMap
     }
 
     public static boolean isEmptyDirectory(String location) throws Exception {
-        String[] files = ImportUtils.getFileSystem(location).getExistingFiles(location, "*");
-        return ((files.length == 0) || (files.length == 1 && "_SUCCESS".equals(truncateFileNameFromFullPath(files[0]))));
-
+        DistributedFileSystem dfs = ImportUtils.getFileSystem(location);
+        return dfs.getInfo(location).isEmptyDirectory();
     }
 
     public static boolean isExisting(String location) throws Exception {
         FileInfo fileInfo = ImportUtils.getFileSystem(location).getInfo(location);
-        return  fileInfo.exists();
+        return fileInfo.exists();
 
     }
+    public static FileInfo getFileInfoOrNull(String location)
+    {
+        try {
+            return ImportUtils.getFileSystem(location).getInfo(location);
+        }
+        catch( StandardException | IOException e )
+        {
+            return null;
```

Comment:
This looks very broad to me. Is there an expected exception we want to catch here? I think in most cases we'd need to bubble up the exception or return a nicer error.

Reply:
ok. i saw there's ImportUtils.getImportFileInfo , and we can use FileInfo.exist() to check for existance, so i removed this function